File: app/src/main/java/com/tunjid/rcswitchcontrol/dialogfragments/NameServiceDialogFragment.java
package com.tunjid.rcswitchcontrol.dialogfragments;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import com.tunjid.rcswitchcontrol.R;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentActivity;

import static java.util.Objects.requireNonNull;


@SuppressLint("InflateParams")
public class NameServiceDialogFragment extends DialogFragment {

    public static NameServiceDialogFragment newInstance() {

        NameServiceDialogFragment fragment = new NameServiceDialogFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    public NameServiceDialogFragment() { }

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        final FragmentActivity activity = requireActivity();
        final ServiceNameListener nameListener = (ServiceNameListener) getParentFragment();
        final LayoutInflater inflater = activity.getLayoutInflater();

        final View view = inflater.inflate(R.layout.dialog_rename_switch, null);
        final EditText editText = view.findViewById(R.id.switch_name);

        return new AlertDialog.Builder(activity, R.style.DialogTheme)
                .setView(view)
                .setTitle(R.string.name_nsd_service)
                .setPositiveButton(R.string.ok, (dialog, id) -> {
                    requireNonNull(nameListener).onServiceNamed(editText.getText().toString());
                    dismiss();
                })
                .setNegativeButton(R.string.cancel, (dialog, id) -> dismiss())
                .create();
    }

    public interface ServiceNameListener {
        void onServiceNamed(String name);
    }
}

File: app/src/main/java/com/tunjid/rcswitchcontrol/dialogfragments/NameServiceDialogFragment.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Adetunji Dahunsi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.tunjid.rcswitchcontrol.dialogfragments

import android.annotation.SuppressLint
import android.app.Dialog
import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.tunjid.rcswitchcontrol.R


@SuppressLint("InflateParams")
class NameServiceDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val activity = requireActivity()
        val nameListener = parentFragment as ServiceNameListener?
        val inflater = activity.layoutInflater

        val view = inflater.inflate(R.layout.dialog_rename_switch, null)
        val editText = view.findViewById<EditText>(R.id.switch_name)

        return AlertDialog.Builder(activity, R.style.DialogTheme)
                .setView(view)
                .setTitle(R.string.name_nsd_service)
                .setPositiveButton(R.string.ok) { _, _ ->
                    nameListener?.onServiceNamed(editText.text.toString())
                    dismiss()
                }
                .setNegativeButton(R.string.cancel) { _, _ -> dismiss() }
                .create()
    }

    interface ServiceNameListener {
        fun onServiceNamed(name: String)
    }

    companion object {

        fun newInstance(): NameServiceDialogFragment {

            val fragment = NameServiceDialogFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}
